controllers/machinehealthcheck_controller_test.go

```diff
@@ -46,906 +45,911 @@ import (
-								Kind:       "InfrastructureMachineTemplate",
-								Name:       infraTmpl.GetName(),
-							},
+						InfrastructureRef: corev1.ObjectReference{
+							APIVersion: "infrastructure.cluster.x-k8s.io/v1alpha3",
+							Kind:       "InfrastructureMachineTemplate",
+							Name:       infraTmpl.GetName(),
 						},
 					},
 				},
+			},
+		}
+		machineSet.Default()
+		g.Expect(testEnv.Create(ctx, machineSet)).To(Succeed())
+
+		// Calculate how many Machines have health check succeeded = false.
+		g.Eventually(func() int {
+			machines := &clusterv1.MachineList{}
+			err := testEnv.List(ctx, machines, client.MatchingLabels{
+				"selector": mhc.Spec.Selector.MatchLabels["selector"],
+			})
+			if err != nil {
+				return -1
 			}
-			machineSet.Default()
-			Expect(testEnv.Create(ctx, machineSet)).To(Succeed())
-
-			// Calculate how many Machines have health check succeeded = false.
-			Eventually(func() int {
-				machines := &clusterv1.MachineList{}
-				err := testEnv.List(ctx, machines, client.MatchingLabels{
-					"selector": mhc.Spec.Selector.MatchLabels["selector"],
-				})
-				if err != nil {
-					return -1
-				}
-				return len(machines.Items)
-			}).Should(Equal(1))
-
-			// Create the MachineHealthCheck instance.
-			mhc.Spec.ClusterName = cluster.Name
-			mhc.Spec.NodeStartupTimeout = &metav1.Duration{Duration: 1 * time.Second}
-			mhc.SetGenerateName("unhealthy-machines-deleted-")
-			Expect(testEnv.Create(ctx, mhc)).To(Succeed())
-
-			// Pause the MachineSet reconciler.
-			machineSetPatch := client.MergeFrom(machineSet.DeepCopy())
-			machineSet.Annotations = map[string]string{
-				clusterv1.PausedAnnotation: "",
-			}
-			Expect(testEnv.Patch(ctx, machineSet, machineSetPatch)).To(Succeed())
-
-			// Calculate how many Machines should be remediated.
-			var unhealthyMachine *clusterv1.Machine
-			Eventually(func() (remediated int) {
-				machines := &clusterv1.MachineList{}
-				err := testEnv.List(ctx, machines, client.MatchingLabels{
-					"selector": mhc.Spec.Selector.MatchLabels["selector"],
-				})
-				if err != nil {
-					return -1
-				}
-
-				for i := range machines.Items {
-					if conditions.Get(&machines.Items[i], clusterv1.MachineOwnerRemediatedCondition) != nil {
-						unhealthyMachine = machines.Items[i].DeepCopy()
-						remediated++
-					}
-				}
-				return
-			}).Should(Equal(1))
-
-			// Unpause the MachineSet reconciler.
-			machineSetPatch = client.MergeFrom(machineSet.DeepCopy())
-			delete(machineSet.Annotations, clusterv1.PausedAnnotation)
-			Expect(testEnv.Patch(ctx, machineSet, machineSetPatch)).To(Succeed())
-
-			// Make sure the Machine gets deleted.
-			Eventually(func() bool {
-				machine := unhealthyMachine.DeepCopy()
-				err := testEnv.Get(ctx, util.ObjectKey(unhealthyMachine), machine)
-				return apierrors.IsNotFound(err) || !machine.DeletionTimestamp.IsZero()
+			return len(machines.Items)
+		}, timeout, 100*time.Millisecond).Should(Equal(1))
```

Comment:
So I just noticed this as I was reviewing another PR. 
I'm guessing either the comment for this block is incorrect or the logic inside the eventually is incorrect and needs to match something similar to line 684.

@ncdc or anyone confirm this?

Reply:
I believe the comment on 779 should be reworded to say we're waiting to make sure the MachineSet controller has created the required Machine from the MachineSet.

I think you can also copy the block from L684 to just before the block at L826